lxd/instance/drivers/driver_qemu.go

```diff
@@ -921,16 +927,45 @@ func (d *qemu) Start(stateful bool) error {
 		"-readconfig", confFile,
 		"-pidfile", d.pidFilePath(),
 		"-D", d.LogFilePath(),
-		"-chroot", d.Path(),
+	}
+
+	// If stateful, restore now.
+	if stateful {
+		if !d.stateful {
+			err = fmt.Errorf("Instance has no existing state to restore")
+			op.Done(err)
+			return err
+		}
+
+		qemuCmd = append(qemuCmd, "-incoming", "defer")
+	} else {
+		// The chroot option can't be used when restoring a migration stream.
```

Comment:
Perhaps use the word "stateful" rather than "migration" here to be consistent with the terminology?

Reply:
Also should we log a warning here to indicate the chroot security mechanism isnt being used due to stateful start?

Reply:
I'll update the comment.

About logging a warning, I don't think so, we normally use warnings to indicate something that's incorrect and that the user can do something about.

In this case, there's nothing that can do about it. From a security point of view, we're at least still running under AppArmor and Seccomp so whole not ideal we at least still have a reasonable safety net.